### ISSUE TITLE
mutable.Specification with Mockito unexpected quit

### ISSUE THREAD

**Issue by kolakao** [etorreborre/specs2#453]
Hi, I'm getting this kind of message:
> This looks like a specs2 exception...
> Please report it with the preceding stacktrace at http://github.com/etorreborre/specs2/issues

When using Specification with Mockito in version 3.7, there is no stacktrace at all,
apart of specs2 there is only one dependency i use:

      <dependency>
            <groupId>io.argonaut</groupId>
            <artifactId>argonaut_2.11</artifactId>
            <version>6.0.4</version>
            <exclusions>
                <exclusion>
                    <groupId>org.scalaz</groupId>
                    <artifactId>scalaz-core_2.11</artifactId>
                </exclusion>
            </exclusions>
        </dependency>

**Comment by etorreborre** [comment]
Having the preceding stacktrace might help indeed :-). Did you include `specs2-mock` dependency?

BTW, I'm writing this from an airport and I might not have any wifi for the next week or so, please be patient.

**Comment by kolakao** [comment]
There was no preceding stacktrace at all... 
I have 
<dependency>
	<groupId>org.specs2</groupId>
	<artifactId>specs2_2.11</artifactId>
	<version>3.7</version>
</dependency>
in pom, this is containing the specs2-mock as well, am i right?

**Comment by etorreborre** [comment]
Yes, this should pull the `specs2-mock` dependency as well. Can you try to simplify your specification so that it doesn't do anything? Something like:
```scala
class MySpec extends org.specs2.Specification with org.specs2.mock.Mockito { def is = s2 """ test $ok """ }
```

And tell me if you have the same issue? The other possibility for getting such a message is to have problematic code in your class body:
```scala
class MySpec extends Specification {
   val a = { sys.error("boom"); 1 }
}
```
The workaround is to use a lazy val:
```scala
class MySpec extends Specification {
   lazy val a = { sys.error("boom"); 1 }
}
```

**Comment by kolakao** [comment]
Ok i got what's wrong... i tried to mock the final class, Still it would be much nicer to have a meaningful message in such case :)
Thank you for your guidance!

**Comment by etorreborre** [comment]
Ok, that's interesting I'll try to see why there was no stack trace in that case.

**Comment by etorreborre** [comment]
I actually get an ok stacktrace when I mock a final class:
```
[error]   CAUSED BY org.mockito.exceptions.base.MockitoException:
[error] Cannot mock/spy class user.Blah
[error] Mockito cannot mock/spy following:
[error]   - final classes
[error]   - anonymous classes
[error]   - primitive types
[error]   org.specs2.mock.MockitoMocker$class.mock(MockitoMocker.scala:21)
[error]   org.specs2.mock.mockito.TheMockitoMocker$$anon$1.mock(TheMockitoMocker.scala:8)
```

with this specification:
```
class TestMutableSpec extends mutable.Specification with Mockito {
  val m = mock[Blah]
  "mocking a final class" >> ok
}
```

Can you create a small project reproducing the issue please?

**Comment by kolakao** [comment]
I will when i have some time :)

**Comment by etorreborre** [comment]
I think I have found the cause for this issue. Please wait for the next `3.7.2` version.

**Comment by etorreborre** [comment]
This issue should be fixed by `3.7.2`. Please re-open if this is not the case.
